homeassistant/components/bluetooth/models.py

```diff
@@ -0,0 +1,107 @@
+"""Models for bluetooth."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final
+
+from bleak import BleakScanner
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import AdvertisementData, AdvertisementDataCallback
+
+from homeassistant.core import CALLBACK_TYPE, callback as hass_callback
+
+_LOGGER = logging.getLogger(__name__)
+
+FILTER_UUIDS: Final = "UUIDs"
+
+HA_BLEAK_SCANNER: HaBleakScanner | None = None
+
+
+class HaBleakScanner(BleakScanner):
+    """BleakScanner that cannot be stopped."""
+
+    _callbacks: list[AdvertisementDataCallback] = []
+
+    @hass_callback
+    def async_register_callback(
+        self, callback: AdvertisementDataCallback, filters: dict[str, Any]
+    ) -> CALLBACK_TYPE:
+        """Register a callback."""
+        self._callbacks.append((callback, filters))
+
+        @hass_callback
+        def _remove_callback():
+            self._callbacks.remove((callback, filters))
+
+        return _remove_callback
+
+    def async_callback_disptacher(
+        self, device: BLEDevice, advertisement_data: AdvertisementData
+    ) -> None:
+        """Dispatch the callback.
+
+        Here we get the actual callback from bleak and dispatch
+        it to all the wrapped HaBleakScannerWrapper classes
+        """
+        for callback, filters in self._callbacks:
+            if (uuids := filters.get(FILTER_UUIDS)) and not uuids.intersection(
+                advertisement_data.service_uuids
+            ):
+                return
```

Comment:
return or continue?

Reply:
continue since there are multiple integrations to check matchers for